code/modules/mob/living/simple_animal/revenant/revenant.dm

```diff
@@ -0,0 +1,134 @@
+			src.visible_message("<b>The revenant</b> lets out a waning screech as violet mist swirls around its dissolving body!")
+			src.icon_state = "revenant_draining"
+			sleep(30)
+			src.death()
+	maxHealth = essence * 2
+	if(!revealed)
+		health = maxHealth //Heals to full when not revealed
+	if(essence < essence_regen_cap && essence_regen)
+		essence += 1
+
+/mob/living/simple_animal/hostile/carp/Process_Spacemove(var/movement_dir = 0)
+	return 1 //Mainly to prevent the no-grav effect
+
+/mob/living/simple_animal/revenant/say(message)
+	return 0 //Revenants cannot speak out loud.
+
+/mob/living/simple_animal/revenant/Stat()
+	..()
+	if(statpanel("Status"))
+		stat(null, "Current essence: [essence]E")
+
+/mob/living/simple_animal/revenant/New()
+	..()
+	notransform = 1
+	spawn(50)
+		if(src.mind)
+			src.mind.spell_list += new /obj/effect/proc_holder/spell/targeted/revenant_harvest
+			src.mind.spell_list += new /obj/effect/proc_holder/spell/targeted/revenant_transmit
+			src.mind.spell_list += new /obj/effect/proc_holder/spell/aoe_turf/revenant_light
+			src.mind.spell_list += new /obj/effect/proc_holder/spell/targeted/revenant_life_tap
+			src.mind.spell_list += new /obj/effect/proc_holder/spell/targeted/revenant_seed_drain
+			src.mind.spell_list += new /obj/effect/proc_holder/spell/targeted/revenant_mindspike
+			src.mind.spell_list += new /obj/effect/proc_holder/spell/targeted/revenant_mindblast
+			notransform = 0
+		else
+			qdel(src)
+
+/mob/living/simple_animal/revenant/death()
+	..(1)
+	src.invisibility = 0
+	visible_message("<span class='danger'>[src] pulses with an eldritch purple light as its form unwinds into smoke.</span>")
+	ghostize()
+	qdel(src)
+	return
+
+/mob/living/simple_animal/revenant/attackby(obj/item/W, mob/living/user, params)
+	..()
+	if(istype(W, /obj/item/weapon/nullrod))
+		src.visible_message("<b>The revenant</b> screeches and flails!", \
+							"<span class='userdanger'>The null rod invokes agony in you! You feel your essence draining away!</span>")
+		src.essence -= 25 //hella effective
+		if(prob(5))
+			src.visible_message("<span class='warning'><b>The revenant is torn apart by the null rod!</b></span>")
+			playsound(src, 'sound/effects/supermatter.ogg', 100, 1)
+			src.death()
+
+
+
+/obj/effect/proc_holder/spell/proc/essence_check(var/essence_cost, var/silent = 0)
+	var/mob/living/simple_animal/revenant/W = usr
+	if(W.essence < essence_cost)
+		if(!silent)
+			W << "<span class='warning'>You need [essence_cost]E to use [name] but you only have [W.essence]E available. Harvest some more things.</span>"
+		return 0
+	W.essence -= essence_cost
+	return 1
+
+
+
+/mob/living/simple_animal/revenant/proc/change_essence_amount(var/essence_amt, var/mode = 0, var/silent = 0, var/source = null, var/mob/living/simple_animal/revenant/user = usr)
+	//Mode 1 is essence subtracted, mode 0 is essence added
+	//Example use: revenant.change_essence_amount(25, 0, 0, "the debug") would tell him "Gained 25E from the debug."
```

Comment:
Can't you just leave out the mode argument and call this with negative amounts of essence?
e.g. change_essence_amount(-25, 0, "the debug") instead of change_essence_amount(25, 1, 0, "the debug")